src/components/widgets/FormWidget/FormWidget.tsx

```diff
@@ -22,20 +23,24 @@ interface FormWidgetProps extends FormWidgetOwnProps {
 }
 
 export const FormWidget: FunctionComponent<FormWidgetProps> = (props) => {
+    const hiddenKeys: string[] = []
     const flattenWidgetFields = useFlatFormFields<WidgetFormField>(props.meta.fields)
+        .filter((item) => {if (item.type !== FieldType.hidden) { return item } else { hiddenKeys.push(item.key); return undefined}})
```

Comment:
It's a tricky way to filter items, mainly due to them not been actually filtered from an array but replaced with `undefined` values, e.g. `['visible', 'visible', undefined, 'visible', undefined, undefined]` vs. `['visible', 'visible', 'visible]`.
How about something like this?
```tsx
const flattenWidgetFields = useFlatFormFields<WidgetFormField>(props.meta.fields)
.filter(item => {
    const isHidden = item.type === FieldType.hidden
    if (isHidden) {
        hiddenKeys.push(item.key)
    }
    return !isHidden
})
```

Reply:
yes, this solution is better. Fixed.